### PR TITLE
Fix OS scan when -V is not set

### PR DESCRIPTION
**Pull request by rphillips**
This PR warns when the os cannot be figured out without the -V option. The default config does not have a valid OS distribution set, since it is version specific.

### CODE REVIEW

**Comment by ashwindasr** [comment]
`cd` (certified distributions) was returning an empty list `[]` if `-V` is not set. This PR aims to fix that by failing with a new error (`certified_distributions is not defined, please use -V [VERSION] on the command line`). For example:

```
| openshift-enterprise-pod-container | /etc/redhat-release | certified_distributions is not defined, please use -V [VERSION] on the command line | registry-proxy.engineering.redhat.com/rh-osbs/openshift-ose-pod-rhel9@sha256:856bcc5a915f5af7dd6bfb35faa54ecb78e350c4ca5adb01d49df8915d3bb258 |
```

File: internal/validations/validations_os.go [hunk 1]
```diff
@@ -22,7 +22,8 @@ func ValidateOS(cfg *types.Config, mountPath string) (info types.OSInfo) {
 	}
 
 	for _, d := range cd {
-		if bytes.HasPrefix(f, []byte(d)) {
+	    cd_bytes = []byte(d)
```
Comment (kolyshkin) [inline internal/validations/validations_os.go]: you have spaces instead of a tab here.

**Comment by ashwindasr** [comment]
also looks like verify is failing, we may need to add 
```
"ErrCertifiedDistributionsEmpty": ErrCertifiedDistributionsEmpty,
```
to `internal/types/error_map.go`

**Comment by kolyshkin** [comment]
Looks like the first two commits needs to be squashed?

File: internal/types/types_scan_result.go [hunk 2]
```diff
@@ -59,10 +58,8 @@ func (r *ScanResult) SetOpenssl(info OpensslInfo) *ScanResult {
 }
 
 func (r *ScanResult) SetOS(info OSInfo) *ScanResult {
-	if errors.Is(info.Error, os.ErrNotExist) {
-		r.SetError(ErrDistributionFileMissing)
```
Comment (rhmdnd) [inline internal/types/types_scan_result.go:63]: These were the only two places using `ErrDistributionFileMissing`. We can probably consider removing the error from errors.go.

Comment (rphillips) [reply internal/types/types_scan_result.go:63]: I added this back in within a different spot. Thanks.

**Comment by rphillips** [comment]
Squashed and fixups the comments.

**Pull request merged**